### PR TITLE
[vectorized][feature](planner)(executor)support grouping sets,rollup,cube(#7599)

### PR DESCRIPTION
**Pull request by anneji-dev**
## Proposed changes

Close related #7599
Support grouping sets,rollup,cube 

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ x] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/function_grouping_id.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+#define DORIS_FUNCTION_GROUPING_ID_H
+
+#include "vec/functions/simple_function_factory.h"
+#include "vec/columns/column_nullable.h"
+#include "vec/functions/function_helpers.h"
+#include "vec/utils/util.hpp"
+#include "vec/data_types/get_least_supertype.h"
+
+namespace doris::vectorized {
+class FunctionGroupingId : public IFunction {
```
Comment (HappenLee) [inline be/src/vec/functions/function_grouping_id.h:28]: the funcion GroupingId and Grouping is similar, you should do a union abstruct. like other funcion maybe the FunctionToNoNullableOneArgument

File: be/src/vec/functions/function_grouping_id.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    bool use_default_implementation_for_nulls() const override { return false; }
+
+    DataTypePtr get_return_type_impl(const ColumnsWithTypeAndName& arguments) const override {
+        return std::make_shared<DataTypeInt64>();
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) override {
+        if (arguments.size() != 1)
```
Comment (HappenLee) [inline be/src/vec/functions/function_grouping_id.h:48]: No Need Check here, you set `get_number_of_arguments()` == 1

File: be/src/vec/functions/function_grouping_id.h [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+        const ColumnWithTypeAndName& src_column = block.get_by_position(arguments[0]);
+        const ColumnWithTypeAndName& rel_column = block.get_by_position(result);
+        if (!src_column.column)
+            return Status::InternalError("Illegal column " + src_column.column->get_name() + " of first argument of function " + name);
+
+        bool src_is_nullable = src_column.type->is_nullable();
+        TypeIndex src_typeid = src_column.type->get_type_id();
+        TypeIndex res_typeid = rel_column.type->get_type_id();
+        if (res_typeid != TypeIndex::Int64)
```
Comment (HappenLee) [inline be/src/vec/functions/function_grouping_id.h:59]: no need check here, FE will promise the argment is bigint

File: be/src/vec/functions/function_grouping_id.h [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            return Status::InternalError("Return type is not int64 of function grouping_id");
+
+        MutableColumnPtr res_column = rel_column.type->create_column();
+        if (src_is_nullable) {
```
Comment (HappenLee) [inline be/src/vec/functions/function_grouping_id.h:63]: the argument may nullable ？ maybe we should promise the argument is not nullable and DCHECK here

File: fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java [hunk 5]
```diff
@@ -1070,6 +1070,16 @@ private PlanNode createRepeatNodePlan(SelectStmt selectStmt, Analyzer analyzer,
...
         root.init(analyzer);
+        // set agg outtuple nullable
+        AggregateInfo aggInfo = selectStmt.getAggInfo();
+        TupleId aggOutTupleId = aggInfo.getOutputTupleId();
+        TupleDescriptor aggOutTupleDescriptor = analyzer.getDescTbl().getTupleDesc(aggOutTupleId);
+        int aggregateExprStartIndex = groupByClause.getGroupingExprs().size();
+        for (int i = 0; i < aggregateExprStartIndex; ++i) {
+            SlotDescriptor slot = aggOutTupleDescriptor.getSlots().get(i);
+            if (!slot.getIsNullable())
+                slot.setIsNullable(true);
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java:1081]: why change the slot nullable ？ give a problem case please

**Review by HappenLee — COMMENTED** [review]
some problem need to recheck

File: be/src/vec/functions/function_grouping_id.h (hunk 3 above)
Comment (anneji-dev) [reply be/src/vec/functions/function_grouping_id.h:59]: ok

File: be/src/vec/functions/function_grouping_id.h (hunk 2 above)
Comment (anneji-dev) [reply be/src/vec/functions/function_grouping_id.h:48]: ok

File: be/src/vec/functions/function_grouping_id.h (hunk 1 above)
Comment (anneji-dev) [reply be/src/vec/functions/function_grouping_id.h:28]: ok

File: be/src/vec/exec/vrepeat_node.cpp [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+        if (is_repeat_slot) {
+            DCHECK(_output_slots[cur_col]->is_nullable());
+            auto* nullable_column = reinterpret_cast<ColumnNullable *>(columns[cur_col].get());
+            auto& null_map = nullable_column->get_null_map_data();
+            auto* column_ptr = columns[cur_col].get();
+
+            // set slot null not in repeat_ids
+            if (is_set_null_slot) {
+                for (size_t j = 0; j < src_column.column->size(); ++j) {
+                    nullable_column->insert_data(nullptr, 0);
```
Comment (HappenLee) [inline be/src/vec/exec/vrepeat_node.cpp:115]: This code maybe can not SIMD? maybe should call resize function to set nullable_column size to SIMD

File: be/src/vec/exec/vrepeat_node.cpp [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+            return Status::OK();
+        }
+    }
+
+    RETURN_IF_ERROR(get_repeated_block(_child_block.get(), _repeat_id_idx, block));
+
+    _repeat_id_idx++;
+
+    int size = _repeat_id_list.size();
+    if (_repeat_id_idx >= size) {
```
Comment (HappenLee) [inline be/src/vec/exec/vrepeat_node.cpp:210]: why here reset the _child_block， call `release_block_memory` to reuse the mem of _child_block

File: be/src/vec/functions/function_grouping.cpp [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "function_grouping.h"
+
+namespace doris::vectorized {
+void register_function_grouping(SimpleFunctionFactory& factory) {
```
Comment (HappenLee) [inline be/src/vec/functions/function_grouping.cpp:21]: void register_function_grouping(SimpleFunctionFactory& factory) {
    factory.register_function<FunctionGrouping>();
    factory.register_function<FunctionGroupingId>();
}

**Review by HappenLee — COMMENTED** [review]
some little problem and rebase code again

**Pull request merged**